packages/cli/src/commands/verify.ts

```diff
@@ -0,0 +1,51 @@
+import { Args, Command, Flags } from '@oclif/core';
+import fs from 'fs/promises';
+import { sigstore } from 'sigstore';
+
+export default class Verify extends Command {
+  static override description = 'describe the command here';
+  static override examples = [
+    '<%= config.bin %> <%= command.id %> ./bundle.sigstore',
+  ];
+  static override enableJsonFlag = true;
+
+  static override flags = {
+    'tlog-threshold': Flags.integer({
+      description: 'number of transparency log entries required to verify',
+      min: 0,
+      default: 1,
+    }),
+    'ctlog-threshold': Flags.integer({
+      description:
+        'number of certificate transparency log entries required to verify',
+      min: 0,
+      default: 1,
+    }),
+  };
+
+  static override args = {
+    bundle: Args.file({
+      description: 'bundle to verify',
+      required: true,
+      exists: true,
+    }),
+  };
+
+  public async run(): Promise<{ verified: boolean }> {
+    const { args, flags } = await this.parse(Verify);
+
+    const options: Parameters<typeof sigstore.verify>[2] = {
+      tlogThreshold: flags['tlog-threshold'],
+      ctLogThreshold: flags['ctlog-threshold'],
+    };
+
+    const bundle = await fs
+      .readFile(args.bundle)
+      .then((data) => JSON.parse(data.toString()));
+
+    return sigstore.verify(bundle, undefined, options).then(() => {
```

Comment:
Do we want to allow passing in `data` for `messageSignature` bundles`? Optional third arg?

Reply:
> Presuming this will exit 1 if verification fails when throwing and then exit 0 on success

Yep.

> Should we add some basic happy/sad path tests around this?

Yeah, I'm gonna come back through and figure out the whole testing strategy for this. There is so much mocking required to do even basic tests here that I decided to punt on this for the first pass.

I'm gonna punt on the messageSignature use case for the moment. I'm wondering if it should be a separate command altogether.